Sources/System/Sockets/SocketAddress+Local.swift

```diff
@@ -0,0 +1,77 @@
+/*
+ This source file is part of the Swift System open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift System project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+*/
+
+private var _pathOffset: Int {
+  // FIXME: If this isn't just a constant, use `offsetof` in C.
+  MemoryLayout<CInterop.SockAddrUn>.offset(of: \CInterop.SockAddrUn.sun_path)!
+}
+
+// @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+extension SocketAddress {
+  public struct Local {
+    internal let _path: FilePath
+
+    public init(_ path: FilePath) {
+      self._path = path
+    }
+
+
+    public init?(_ address: SocketAddress) {
```

Comment:
Add some generic comment for all of these, something like
```
    //// If `address` represents a [`Self`], create it, otherwise return `nil`
```

Also, should these be inits or optional properties on `SocketAddress`?